### ISSUE TITLE
LinkageError: com.sun.source.tree.CaseTree.CaseKind with Refaster on Java 17

### ISSUE THREAD

**Issue by PhilippWendler** [google/error-prone#3553]
While attempting to run Refaster on Java 17, I encounter the following stack trace:
```
    [javac]   java.lang.LinkageError: com.sun.source.tree.CaseTree.CaseKind                                                                                            
    [javac]     at com.google.errorprone.refaster.PlaceholderUnificationVisitor.makeCase(PlaceholderUnificationVisitor.java:696)                                          
    [javac]     at com.google.errorprone.refaster.PlaceholderUnificationVisitor.lambda$visitCase$90(PlaceholderUnificationVisitor.java:665)                               
    [javac]     at com.google.errorprone.refaster.PlaceholderUnificationVisitor.lambda$chooseSubtrees$8(PlaceholderUnificationVisitor.java:316)   

[...snipped...]

    [javac]   Caused by: java.lang.ClassNotFoundException: com.sun.source.tree.CaseTree.CaseKind
    [javac]     at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
    [javac]     at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
    [javac]     at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
    [javac]     at java.base/java.lang.Class.forName0(Native Method)
    [javac]     at java.base/java.lang.Class.forName(Class.java:375)
    [javac]     at com.google.errorprone.refaster.PlaceholderUnificationVisitor.makeCase(PlaceholderUnificationVisitor.java:679)
    [javac]     ... 131 more
```

So far I didn't create a minimal reproducer because I think the bug is relatively clear in line 679:
https://github.com/google/error-prone/blob/3b266369082a26e59940828cb4d8bd264a13cd31/core/src/main/java/com/google/errorprone/refaster/PlaceholderUnificationVisitor.java#L679

[CaseKind](https://github.com/openjdk/jdk/blob/dfacda488bfbe2e11e8d607a6d08527710286982/src/jdk.compiler/share/classes/com/sun/source/tree/CaseTree.java#L121-L130) is a nested enum of the `CaseTree` interface, so that string should be `com.sun.source.tree.CaseTree$CaseKind`.

Alternatively, a few lines above is this code:
https://github.com/google/error-prone/blob/3b266369082a26e59940828cb4d8bd264a13cd31/core/src/main/java/com/google/errorprone/refaster/PlaceholderUnificationVisitor.java#L671
The `getCaseKind()` method returns an instanceof `CaseKind`, so `caseKind.getClass()` would also be a possible way to retrieve the desired `CaseKind.class` object.

If contrary to my current assumption more information about my environment or a reproducer is needed, I will be glad to provide it.

**Comment by PhilippWendler** [comment]
Given that the fix should be quite easy, is there a chance for this to be in the next release of Error Prone? That would be highly appreciated!

**Comment by rickie** [comment]
Hi @PhilippWendler, I'm interesting in picking this up and creating a PR.

Do you maybe have a minimal reproducible example to work with?

**Comment by PhilippWendler** [comment]
The bug triggers (at least) if there is a `@Placeholder` call in the rule for which the match contains a `switch`.

Example rule:
```java
import com.google.errorprone.refaster.annotation.AfterTemplate;
import com.google.errorprone.refaster.annotation.BeforeTemplate;
import com.google.errorprone.refaster.annotation.MayOptionallyUse;
import com.google.errorprone.refaster.annotation.Placeholder;

abstract class Rule {

  @Placeholder
  abstract void placeholder(@MayOptionallyUse String s);

  @BeforeTemplate
  void before(String s) {
    if (s.isEmpty()) {
      placeholder(s);
    }
  }

  @AfterTemplate
  void after(String s) {
    if (s.isEmpty()) {
      placeholder(s);
    }
  }
}
```
Code:
```java
public class Test {

  public static void main(String[] args) {
    if (args[0].isEmpty()) {
      switch (args[0].length()) {
        case 0:
          System.out.println("");
      }
    }
  }
}
```

**Comment by PhilippWendler** [comment]
Seems this should be fixed by #3739 / 68198e71ac58c6fce9592c9588edd4a6aaa3550a. Thanks!
